core/document/src/operation.rs

```diff
@@ -1,4 +1,6 @@
 pub enum Operation {
-	AddCircle { cx: f64, cy: f64, r: f64 },
-	AddRect { x0: f64, y0: f64, x1: f64, y1: f64 },
+	AddCircle { path: String, cx: f64, cy: f64, r: f64 },
+	AddRect { path: String, x0: f64, y0: f64, x1: f64, y1: f64 },
+	DeleteElement { path: String },
+	AddFolder { path: String },
```

Comment:
We'll also need an index for where (top, bottom, somewhere in the middle) to insert the circle/rect/folder within the directory specified by the `path`.